app/experimenter/experiments/tests/test_models/test_models_nimbus.py

```diff
@@ -352,6 +352,54 @@ def test_should_end_returns_True_after_proposed_end_date(self):
         ).update(changed_on=datetime.datetime.now() - datetime.timedelta(days=10))
         self.assertTrue(experiment.should_end)
 
+    def test_computed_enrollment_days_returns_changed_on_minus_start_date(self):
+        # simulates an experiment that was started and ended on the same day
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
+        )
+
+        self.assertEqual(
+            experiment.computed_enrollment_days,
+            0,
+        )
+
+    def test_computed_enrollment_days_returns_fallback(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED
+        )
+
+        self.assertEqual(
+            experiment.computed_enrollment_days,
+            experiment.proposed_enrollment,
+        )
+
+    def test_computed_duration_returns_computed_end_date_minus_start_date(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
+            proposed_duration=10,
+        )
+
+        experiment.changes.filter(
+            old_status=NimbusExperiment.Status.DRAFT,
+            new_status=NimbusExperiment.Status.LIVE,
+        ).update(changed_on=datetime.datetime.now() - datetime.timedelta(days=7))
+
+        # the number of days rounds down
+        self.assertEqual(
+            experiment.computed_duration,
+            6,
+        )
+
+    def test_computed_duration_returns_fallback(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+        )
+
+        self.assertEqual(
+            experiment.computed_duration,
+            experiment.proposed_duration,
+        )
+
```

Comment:
Oh I see we're missing a test that hits the case where the experiment launched and has a real start date, but hasn't ended so the end date is computed.

Reply:
I went ahead and added `computed_end_date` tests to the model.